### PR TITLE
[Relay-TFLite] FP32 and Quantized Object Detection Model

### PR DESCRIPTION
**Pull request by anijain2305**
This PR does multiple things

* **FP32 object detection model** - Adding test for TFLite FP32 object detection. The model is uploaded to dmlc/web-data with this PR - https://github.com/dmlc/web-data/pull/247
* **Quantized object detection model**
   * **Fused activations** - Added the code for fused activations. It calculates the clip off points given scale and zero point
   * **End to end test** - Added the TFLite hosted quantized coco model - https://www.tensorflow.org/lite/models/object_detection/overview
* **Flexbuffer parsing** - Flexbuffer parsing is improved

#### Caveats
* Though TVM and TFLite output for quantized model is close (all bounding boxes are predicted correctly with close probability scores and bounding box locations), they are not exactly equal. Therefore, I had to lower the tolerance to 1e-2. This is because of differences in TFLite and TVM rounding which is getting tackled here - https://github.com/apache/incubator-tvm/pull/4828. Additionally, the output tensors are sorted based on scores, but there can be different TVM and TFLite ordering when multiple bounding boxes have exactly the same score (typically for low scores). Therefore, the test compares only those bounding boxes which have score > 0.6 (a threshold is typically used in end to end applications to discard low score bounding boxes)

* We cannot run the quantized TFLite test in CI due to a known LLVM bug - https://discuss.tvm.ai/t/segfault-in-llvm/3567. We can resolve this by adding device flags like `mcpu=core-avx2', but this does not work well with CI. Therefore, the test is skipped for now.

### CODE REVIEW

**Comment by anijain2305** [comment]
@mbaret @FrozenGene @u99127 @siju-samuel 

Please review.

File: python/tvm/relay/frontend/tflite.py [hunk 1]
```diff
@@ -320,6 +321,45 @@ def dequantize(self, expr, tensor):
                                          input_zero_point=tensor.qnn_params['zero_point'])
         return dequantized
 
+
+    def convert_qnn_fused_activation_function(self, expr, fused_activation_fn,
+                                              scale, zero_point, dtype):
+        """Convert TFLite fused activation function. The expr is an input quantized tensor with
+        scale and zero point """
```
Comment (mbaret) [inline python/tvm/relay/frontend/tflite.py:328]: I think this could be a separate PR as it's not specific to object detection.

Comment (anijain2305) [reply python/tvm/relay/frontend/tflite.py:328]: This is needed for quantized object detection model

**Comment by anijain2305** [comment]
@FrozenGene Please review
@mbaret Please take a second look

File: python/tvm/relay/testing/tf.py [hunk 2]
```diff
@@ -184,10 +184,15 @@ def get_workload_official(model_url, model_sub_path):
     dir_path = os.path.dirname(model_path)
 
     import tarfile
+    import zipfile
```
Comment (siju-samuel) [inline python/tvm/relay/testing/tf.py:187]: suggest to move this imports inside if or elif. let it import based on file type.

File: tests/python/frontend/tflite/test_forward.py [hunk 3]
```diff
@@ -1942,6 +1960,100 @@ def test_forward_qnn_mobilenet_v3_net():
     tvm.testing.assert_allclose(tvm_sorted_labels, tflite_sorted_labels)
 
 
+#######################################################################
+# Quantized SSD Mobilenet
+# -------------
```
Comment (siju-samuel) [inline tests/python/frontend/tflite/test_forward.py:1965]: nit , align the dashes

File: python/tvm/relay/frontend/tflite.py (hunk 1 above)
Comment (mbaret) [reply python/tvm/relay/frontend/tflite.py:328]: It's more from a position of having the history clearly reflect when features were added. Supporting fused qnn functions is needed for this model, but it's also necessary for a wide range of other models so I'd view this as a feature not specific to object detection.

**Comment by anijain2305** [comment]
Ping @FrozenGene  :)

**Comment by FrozenGene** [comment]
THANKS @anijain2305 @siju-samuel @mbaret
